application/forms/ProcessForm.php

```diff
@@ -212,4 +237,19 @@ public function onSuccess()
 
         parent::onSuccess();
     }
+
+    public function hasDeleteButton()
+    {
+        return $this->deleteButtonName !== null;
+    }
+
+    public function shouldBeDeleted()
+    {
+        if (! $this->hasDeleteButton()) {
+            return false;
+        }
+
+        $name = $this->deleteButtonName;
+        return $this->getSentValue($name) === $this->getElement($name)->getLabel();
```

Comment:
$name === $label.

Reply:
You used $label as the element name above (this may change with the previously requested changes). So `$this->getElement($name)->getLabel()` === `$name`.

Reply:
But the label will still remain `$this->translate('Delete')`